src/Feature/Editor/GutterView.re

```diff
@@ -0,0 +1,201 @@
+open Revery.UI;
+open Oni_Core;
+
+module Constants = {
+  include Constants;
+
+  let diffMarkersMaxLineCount = 2000;
+  let diffMarkerWidth = 3.;
+  let gutterMargin = 3.;
+};
+
+let renderLineNumber =
+    (
+      ~context: Draw.context,
+      lineNumber: int,
+      lineNumberWidth: float,
+      theme: Theme.t,
+      lineSetting,
+      cursorLine: int,
+      yOffset: float,
+    ) => {
+  let isActiveLine = lineNumber == cursorLine;
+  let y = yOffset -. context.fontMetrics.ascent;
+
+  let lineNumber =
+    string_of_int(
+      LineNumber.getLineNumber(
+        ~bufferLine=lineNumber + 1,
+        ~cursorLine=cursorLine + 1,
+        ~setting=lineSetting,
+        (),
+      ),
+    );
+
+  let lineNumberXOffset =
+    isActiveLine
+      ? 0.
+      : lineNumberWidth
+        /. 2.
+        -. float(String.length(lineNumber))
+        *. context.charWidth
+        /. 2.;
+
+  let lineNumberTextColor =
+    isActiveLine
+      ? theme.editorActiveLineNumberForeground
+      : theme.editorLineNumberForeground;
+  let paint = Skia.Paint.make();
+  Skia.Paint.setColor(paint, Revery.Color.toSkia(lineNumberTextColor));
+  Skia.Paint.setTextEncoding(paint, Utf8);
```

Comment:
These calls are actually a bit expensive since they haven't been optimized (ie, with `[@noalloc]`), which is why they were factored out previously